src/batched-json.js

```diff
@@ -27,24 +37,49 @@ import {getValueForExpr} from './json';
  * @param {!Element} element
  * @param {string=} opt_expr Dot-syntax reference to subdata of JSON result
  *     to return. If not specified, entire JSON result is returned.
+ * @param {UrlReplacementPolicy=} opt_urlReplacement If ALL, replaces all URL vars.
+ *     If OPT_IN, replaces whitelisted URL vars. Otherwise, don't expand.
  * @return {!Promise<!JsonObject|!Array<JsonObject>>} Resolved with JSON
  *     result or rejected if response is invalid.
  */
-export function fetchBatchedJsonFor(ampdoc, element, opt_expr) {
+export function batchFetchJsonFor(
+  ampdoc,
+  element,
+  opt_expr = '.',
+  opt_urlReplacement = UrlReplacementPolicy.NONE)
+{
   const url = assertHttpsUrl(element.getAttribute('src'), element);
-  return Services.urlReplacementsForDoc(ampdoc).expandUrlAsync(url)
-      .then(src => {
-        const opts = {};
-        if (element.hasAttribute('credentials')) {
-          opts.credentials = element.getAttribute('credentials');
-        } else {
-          opts.requireAmpResponseSourceOrigin = false;
-        }
-        return Services.batchedXhrFor(ampdoc.win).fetchJson(src, opts);
-      }).then(res => res.json()).then(data => {
-        if (data == null) {
-          throw new Error('Response is undefined.');
+
+  // Replace vars in URL if desired.
+  const urlReplacements = Services.urlReplacementsForDoc(ampdoc);
+  const srcPromise = (opt_urlReplacement >= UrlReplacementPolicy.OPT_IN)
```

Comment:
In the `none` case, should they be replaced with ``? leaving them as is would generate strange and wrong Urls.

Reply:
Leaving it unmodified is the default behavior for other opt-ins, e.g. `<a>`.